### DISCUSSION TITLE
Added TSNAccessPoint for supporting TSN in wireless by building on 80211 interface. How to contribute it back to INET

### DISCUSSION THREAD

**Discussion by Abh4git** [inet-framework/inet#1011] [Ideas]
As part of my PhD work I have created a TSNAccessPoint by building on 802.11 interface. Is it already included or planned. Can I contribute it back to INET framework? If so how should I go forward?
https://link.springer.com/article/10.1007/s42979-024-03475-z

Thanks
Regards
Abhilash

**Comment by rhornig** [comment]
We don't have anything related to TSN and wireless integration (even in plans) so contributions are welcome. AFAIK, there is an ongoing effort to better integrate Simu5G and TSN switches, but that is on the wired part of the Simu5G model. We would be happy to have something like this as this topic seems to be on cutting edge. Currently we are working towards the INET 4.6 release (based on the master branch). If you could send a pull request against the master branch, we could start discussing the plans.

**Comment by Abh4git** [comment]
Thank you! will do that.

**Comment by Abh4git** [comment]
As I understand, I need to create a fork and raise pull request. Am I right about the process?

**Reply by levy** [reply]
Yes, submitting a pull request is the right way to change code in INET.

**Comment by Abh4git** [comment]
Thank you! I have done that. I have an example as well with the TsnAccessPoint which could be used to showcase. But for getting it working I had few workaround changes in INET code. That part I will discuss in detail and get it corrected before submitting separate from this.
